src/mlpack/methods/decision_tree/mse_gain.hpp

```diff
@@ -41,10 +41,10 @@ class MSEGain
    * @param end End index.
    */
   template<bool UseWeights, typename VecType, typename WeightVecType>
-  static double Evaluate(const VecType& values,
```

Comment:
Was there a reason to change these from `static`?  I think they should work just fine as they were---let me know if I overlooked something.

Reply:
I think we can remove this function and move its content to the overload just below it. Because we are using this function only in line https://github.com/RishabhGarg108/mlpack-1/blob/a4081bf0c8029f70b1a9d0a36851d3285fccd43d/src/mlpack/methods/decision_tree/mse_gain.hpp#L95.

Regarding it being non-static, I kept it this way because the place at which this function is called ([here](https://github.com/RishabhGarg108/mlpack-1/blob/a4081bf0c8029f70b1a9d0a36851d3285fccd43d/src/mlpack/methods/decision_tree/decision_tree_regressor_impl.hpp#L625)), we are already using an instantiated `FitnessFunction`. So it won't matter if it is static or not.

If you like then I can make it static back again. No harm in that :) Let me know what do you think 👍

Reply:
Yeah, you are right, it doesn't matter, but I do think it keeps the tests a bit simpler.  So I would suggest changing it, but, in the end, up to you. :+1: